### PR TITLE
highlights(jsx): fix constant identifier

### PR DESCRIPTION
**Pull request by MunifTanjim**
**Before**:

image

**After**:

image

### CODE REVIEW

File: queries/jsx/highlights.scm [hunk 1]
```diff
@@ -32,7 +32,4 @@
 ; Handle the dot operator effectively - <My.Component />
 (jsx_self_closing_element ((nested_identifier (identifier) @tag (identifier) @constructor)))
 
-(variable_declarator ((identifier) @type
- (#match? @type "^[A-Z]")))
```
Comment (stsewd) [inline queries/jsx/highlights.scm:36]: I think this query is meant to stay, you make constant have more priority, you can move it down this query.

Comment (MunifTanjim) [reply queries/jsx/highlights.scm:36]: Why does `jsx` have `@type`? 🤔 It isn't even `tsx`.

The constant comes from `javascript`. So, I don't think changing priority can be done... can it be?

Comment (stsewd) [reply queries/jsx/highlights.scm:36]: > Why does jsx have @type? thinking It isn't even tsx.

Not a js dev, so can't answer that p: /cc @steelsojka 

> The constant comes from javascript. So, I don't think changing priority can be done... can it be?

Think you can just duplicate the query

Comment (MunifTanjim) [reply queries/jsx/highlights.scm:36]: > Not a js dev, so can't answer that p:

JSX files have no types. I think that was a mistake. Most of the other mistakenly added `@type`s were removed in https://github.com/nvim-treesitter/nvim-treesitter/pull/1357. This one is the leftover.

Comment (antonk52) [reply queries/jsx/highlights.scm:36]: I think I came across this while creating #1547, generally matching based on the casing doesn't seem like a good idea. Also this PR seem well reasonable

Comment (steelsojka) [reply queries/jsx/highlights.scm:36]: Type in this case isn't specifically a type like in typescript... but I do think it is miss marked. We shouldn't be making assumptions on what this variable is. We should only highlight StartCase components that are used as JSX components... so I think this is safe to remove.

**Review by steelsojka — APPROVED** [review]

**Pull request merged**